x-pack/plugins/cases/public/components/visualizations/actions/utils.ts

```diff
@@ -26,12 +27,14 @@ type PersistableStateAttachmentWithoutOwner = Omit<PersistableStateAttachmentPay
 export const getLensCaseAttachment = ({
   timeRange,
   attributes,
+  metadata,
 }: {
-  timeRange: LensEmbeddableInput['timeRange'];
+  timeRange: LensProps['timeRange'];
   attributes: LensSavedObjectAttributes;
+  metadata?: LensProps['metadata'];
 }): PersistableStateAttachmentWithoutOwner =>
   ({
-    persistableStateAttachmentState: { attributes, timeRange },
+    persistableStateAttachmentState: { attributes, timeRange, metadata },
```

Comment:
Maybe we can avoid persisting the `metadata` if it is `undefined` or set it to `{}` if it is empty. Wdyt?

Reply:
I think that it doesn't store properties that are undefined. I tested it by attaching other visualisations to cases and the response doesn't contain `metadata`. 
![Screenshot 2024-01-29 at 15 03 26](https://github.com/elastic/kibana/assets/1490444/63787044-f654-49b9-a9c7-53d31b6c00a6)


btw, `timeRange` is also optional, and we don't remove it.

Reply:
Good point! Thanks for checking out, I was not aware.